pkg/occlient/occlient.go

```diff
@@ -1137,37 +1134,7 @@ func (c *Client) BootstrapSupervisoredS2I(params CreateArgs, commonObjectMeta me
 	}
 
 	// Append s2i related parameters extracted above to env
-	inputEnvs = uniqueAppendOrOverwriteEnvVars(
-		inputEnvs,
-		corev1.EnvVar{
-			Name:  EnvS2IScriptsURL,
-			Value: s2iPaths.ScriptsPath,
-		},
-		corev1.EnvVar{
-			Name:  EnvS2IScriptsProtocol,
-			Value: s2iPaths.ScriptsPathProtocol,
-		},
-		corev1.EnvVar{
-			Name:  EnvS2ISrcOrBinPath,
-			Value: s2iPaths.SrcOrBinPath,
-		},
-		corev1.EnvVar{
-			Name:  EnvS2IDeploymentDir,
-			Value: s2iPaths.DeploymentDir,
-		},
-		corev1.EnvVar{
-			Name:  EnvS2IWorkingDir,
-			Value: s2iPaths.WorkingDir,
-		},
-		corev1.EnvVar{
-			Name:  EnvS2IBuilderImageName,
-			Value: s2iPaths.BuilderImgName,
-		},
-		corev1.EnvVar{
-			Name:  EnvS2IDeploymentBackupDir,
-			Value: DefaultS2IDeploymentBackupDir,
-		},
-	)
+	inputEnvs = uniqueUpsertEnvVarsFromS2IPaths(inputEnvs, s2iPaths)
```

Comment:
what's `Upsert`? Isn't that related to databases? :)

Reply:
but I think that conveys the meaning without being too verbose